src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs

```diff
@@ -138,410 +144,589 @@ private bool TryGetValueFromDocument(BlittableObjectInstance parent, string prop
                     {
-                        case Client.Constants.Documents.Indexing.Fields.NullValue:
-                            value = DynamicJsNull.ExplicitNull;
-                            return true;
+                        if (fieldReader.Read(out var valueInEntry) == false)
+                            goto case IndexEntryFieldType.Invalid;
 
-                        case Client.Constants.Documents.Indexing.Fields.EmptyString:
-                            value = string.Empty;
-                            return true;
+                        value = Encodings.Utf8.GetString(valueInEntry);
                     }
-                }
+                        break;
 
-                if (fieldType.IsNumeric)
-                {
-                    if (long.TryParse(val, NumberStyles.Integer, CultureInfo.InvariantCulture, out var valueAsLong))
+                    case IndexEntryFieldType.ListWithNulls:
+                    case IndexEntryFieldType.List:
                     {
-                        value = valueAsLong;
+                        if (fieldReader.TryReadMany(out var iterator) == false)
+                            goto case IndexEntryFieldType.Invalid;
+
+                        //todo PERF
+                        List<string> array = new(iterator.Count);
+                        while (iterator.ReadNext())
+                        {
+                            if (iterator.IsEmptyString || iterator.IsNull)
+                            {
+                                array.Add(iterator.IsNull ? null : string.Empty);
+                            }
+                            else
+                            {
+                                array.Add(Encodings.Utf8.GetString(iterator.Sequence));
+                            }
+                        }
+
+                        value = FromObject(parent.Engine, array);
                     }
-                    else if (double.TryParse(val, NumberStyles.Any, CultureInfo.InvariantCulture, out var valueAsDouble))
+                        break;
+                    case IndexEntryFieldType.RawList:
                     {
-                        value = valueAsDouble;
+                        if (fieldReader.TryReadMany(out var iterator) == false)
+                            goto case IndexEntryFieldType.Invalid;
+
+                        JsValue[] arrayItems = new JsValue[iterator.Count];
+                        var idX = 0;
+                        while (iterator.ReadNext())
+                        {
+                            fixed (byte* ptr = &iterator.Sequence.GetPinnableReference())
+                            {
+                                var itemAsBlittable = new BlittableJsonReaderObject(ptr, iterator.Sequence.Length, parent.Blittable._context);
+                                arrayItems[idX] = TranslateToJs(parent, indexField.Name, BlittableJsonToken.StartObject, itemAsBlittable);
```

Comment:
Is it okay that `ptr` will no longer be pinned? Can an item in `arrayItems` still reference that pointer?

Reply:
@redknightlois  when we take an entry from the container, the pointer is valid until we close the transaction, right?

Reply:
It that was the case then why we'd need to wrap it by `fixed` ?

Reply:
okay